test/services/account/services/accounts.test.js

```diff
@@ -581,6 +581,102 @@ describe('Account Service', () => {
 					done();
 				});
 		});
+
+		it('should not allow external request when the requester and the requested user are not from the same school', async () => {
+			const school = await testObjects.createTestSchool({
+				name: 'testSchool1',
+			});
+			const otherSchool = await testObjects.createTestSchool({
+				name: 'testSchool2',
+			});
+
+			const student = await testObjects.createTestUser({ roles: ['student'], schoolId: school._id });
+			const studentAccount = await testObjects.createTestAccount(
+				{ username: student.email, password: student.email },
+				undefined,
+				student
+			);
+
+			const user = await testObjects.createTestUser({ roles: ['teacher'], schoolId: otherSchool._id });
+			const params = await generateRequestParamsFromUser(user);
+
+			expect(student.schoolId).to.not.equal(user.schoolId);
+
+			try {
+				await accountService.find({ ...params, query: { userId: studentAccount.userId } });
+				expect.fail('The previous call should have failed');
+			} catch (err) {
+				expect(err.code).to.equal(403);
+				expect(err.message).to.equal('You are not allowed to request this information');
+			}
+		});
+
+		it('should allow external request when the requester and the requested user are from the same school', async () => {
+			const student = await testObjects.createTestUser({ roles: ['student'] });
+			const studentAccount = await testObjects.createTestAccount(
+				{ username: student.email, password: student.email },
+				undefined,
+				student
+			);
+
+			const user = await testObjects.createTestUser({ roles: ['teacher'] });
+			const params = await generateRequestParamsFromUser(user);
+
+			expect(student.schoolId.toString()).to.equal(user.schoolId.toString());
+
+			const requestedAccount = await accountService.find({ ...params, query: { userId: studentAccount.userId } });
+			expect(requestedAccount).to.not.have.lengthOf(0);
```

Comment:
You should check if it's really the account you requested and not random data, just to make sure